src/Microsoft.ML.Data/DataLoadSave/Text/TextLoader.cs

```diff
@@ -92,10 +98,18 @@ public Column(string name, DataKind? type, Range[] source, KeyCount keyCount = n
             public string Name;
 
             /// <summary>
-            /// <see cref="DataKind"/> of the items in the column. If <see langword="null"/> defaults to a float.
+            /// <see cref="InternalDataKind"/> of the items in the column. If <see langword="null"/> defaults to a float.
+            /// Although <see cref="InternalDataKind"/> is internal, <see cref="Type"/>'s information can be publically accessed by <see cref="ItemType"/>.
             /// </summary>
             [Argument(ArgumentType.AtMostOnce, HelpText = "Type of the items in the column")]
-            public DataKind? Type;
+            [BestFriend]
+            internal InternalDataKind? Type;
+
+            /// <summary>
+            /// <see cref="DataKind"/> of the items in the column.
+            /// </summary>
+            /// It's a public interface to access the information in an internal DataKind.
+            public DataKind ItemType => Type.HasValue ? Type.Value.ToScalarType() : DataKind.Single;
```

Comment:
`ToScalarType()`?

Reply:
Will rename it to ToDataKind.

---
In reply to: [259519970](https://github.com/dotnet/machinelearning/pull/2661#discussion_r259519970) [](ancestors = 259519970)